tests/cli/cli.test.ts

```diff
@@ -0,0 +1,398 @@
+    txHash = extractFromStdout(stdout, TX_HASH_REGEX);
+
+    expect(txFeeEth).to.not.be.undefined;
+    expect(txFeeWei).to.not.be.undefined;
+
+    expect(contract_class_hash).to.not.be.undefined;
+    expect(txHash).to.not.be.undefined;
+  });
+
+  this.afterAll('Declare transaction status', async () => {
+    const res = await sh(
+      `${warpBin} status ${txHash} --network ${network} --gateway_url ${gatewayURL} --feeder_gateway_url ${gatewayURL}`,
+    );
+
+    expect(res.stderr).to.be.empty;
+    expect(res.stdout).to.not.be.empty;
+
+    const { tx_status } = JSON.parse(res.stdout);
+    expect(tx_status).to.equal('ACCEPTED_ON_L2');
+  });
+});
+
+describe('Deploy the MyToken contract', function () {
+  this.timeout(TIME_LIMIT);
+
+  let txHash: string;
+
+  it('should deploy the ERC20 contract', async () => {
+    const { stdout, stderr } = await sh(
+      `${warpBin} deploy ${contractCairoFile} --inputs 500 0 --wallet ${wallet} --feeder_gateway_url ${gatewayURL} --gateway_url ${gatewayURL} --network ${network} --account_dir ${accountDir}`,
+    );
+
+    expect(stderr).to.be.empty;
+
+    const txFeeEth = extractFromStdout(stdout, TX_FEE_ETH_REGEX);
+    const txFeeWei = extractFromStdout(stdout, TX_FEE_WEI_REGEX);
+    contract_address = extractFromStdout(stdout, CONTRACT_ADDRESS_REGEX);
+    txHash = extractFromStdout(stdout, TX_HASH_REGEX);
+
+    expect(txFeeEth).to.not.be.undefined;
+    expect(txFeeWei).to.not.be.undefined;
+
+    expect(contract_class_hash).to.not.be.undefined;
+    expect(txHash).to.not.be.undefined;
+  });
+
+  this.afterAll('Deploy transaction status', async () => {
+    const res = await sh(
+      `${warpBin} status ${txHash} --network ${network} --gateway_url ${gatewayURL} --feeder_gateway_url ${gatewayURL}`,
+    );
+
+    expect(res.stderr).to.be.empty;
+    expect(res.stdout).to.not.be.empty;
+
+    const response = JSON.parse(res.stdout);
+    expect(response.tx_status).to.equal('ACCEPTED_ON_L2');
+  });
+});
+
+describe('Call MyToken contract functions', function () {
+  this.timeout(TIME_LIMIT);
+
+  it('get Total Supply', async () => {
+    expect(contract_address).to.not.be.undefined;
+    const { stdout, stderr } = await sh(
+      `${warpBin} call ${contractCairoFile} --function totalSupply --address ${contract_address} --network ${network} --gateway_url ${gatewayURL} --feeder_gateway_url ${gatewayURL} --wallet ${wallet} --account_dir ${accountDir}`,
+    );
+    expect(stderr).to.be.empty;
+    expect(stdout.split('\n')[1].trim()).to.be.equal('500');
+  });
+
+  it('get Total Supply call using cairo ABI', async () => {
+    expect(contract_address).to.not.be.undefined;
+    const { stdout, stderr } = await sh(
+      `${warpBin} call ${contractCairoFile} --use_cairo_abi --function totalSupply_18160ddd --address ${contract_address} --network ${network} --gateway_url ${gatewayURL} --feeder_gateway_url ${gatewayURL} --wallet ${wallet} --account_dir ${accountDir}`,
```

Comment:
Remove --use_cairo_abi; that is only meant for testing.
Using this will not test the encoding of the Solidity inputs

Reply:
done!